diff --git a/dist/module.cjs b/dist/module.cjs
index 13312f85f1bec7282f8901afc480a249b8c1bf85..2f0137dbe4361c2c91a24bfdb59dcae635b07301 100644
--- a/dist/module.cjs
+++ b/dist/module.cjs
@@ -121,7 +121,10 @@ const module$1 = kit.defineNuxtModule({
         getContents: async () => [
           `import { defineNuxtPlugin, useHead } from '#imports'`,
           `const css = \`${(await css).replace(/\s+/g, " ")}\``,
-          `export default defineNuxtPlugin(() => { useHead({ style: [{ children: css ${!nuxt.options.dev && options.inject ? "+ __INLINED_CSS__ " : ""}}] }) })`
+          `export default defineNuxtPlugin({`,
+          `  env: { islands: false },`,
+          `  setup() { useHead({ style: [{ children: css ${!nuxt.options.dev && options.inject ? "+ __INLINED_CSS__ " : ""}}] }) },`,
+          `})`,
         ].join("\n"),
         mode: "server"
       });
diff --git a/dist/module.mjs b/dist/module.mjs
index ea506215aa190c8037d49aaa34b7a5636040ed4e..a483855b5e22a3ae2010a7b5819e3977d86e3e85 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -115,7 +115,10 @@ const module = defineNuxtModule({
         getContents: async () => [
           `import { defineNuxtPlugin, useHead } from '#imports'`,
           `const css = \`${(await css).replace(/\s+/g, " ")}\``,
-          `export default defineNuxtPlugin(() => { useHead({ style: [{ children: css ${!nuxt.options.dev && options.inject ? "+ __INLINED_CSS__ " : ""}}] }) })`
+          `export default defineNuxtPlugin({`,
+          `  env: { islands: false },`,
+          `  setup() { useHead({ style: [{ children: css ${!nuxt.options.dev && options.inject ? "+ __INLINED_CSS__ " : ""}}] }) },`,
+          `})`,
         ].join("\n"),
         mode: "server"
       });
