diff --git a/dist/index.cjs b/dist/index.cjs
index 6598e2cb714aec657906d76c82a6adf4d6c1c960..078f8a47bdf26fd8965638677045e76f887bae5b 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -8,7 +8,7 @@ var _core = require('@tiptap/core');
 
 
 var _state = require('@tiptap/pm/state');
-var _debounce = require('lodash/debounce'); var _debounce2 = _interopRequireDefault(_debounce);
+var _debounce = require('lodash/debounce.js'); var _debounce2 = _interopRequireDefault(_debounce);
 var _tippyjs = require('tippy.js'); var _tippyjs2 = _interopRequireDefault(_tippyjs);
 var BubbleMenuView = class {
   constructor({
diff --git a/dist/index.js b/dist/index.js
index 1e34dac3cb871bb44eee6091b313757e60609aa8..5bb0cb562dff5008f6baecf74072c29605b007d1 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -8,7 +8,7 @@ import {
   posToDOMRect
 } from "@tiptap/core";
 import { Plugin, PluginKey } from "@tiptap/pm/state";
-import debounce from "lodash/debounce";
+import debounce from "lodash/debounce.js";
 import tippy from "tippy.js";
 var BubbleMenuView = class {
   constructor({
