diff --git a/dist/index.cjs b/dist/index.cjs
index b73fd4c58d377b9c584035cb106106011ad68e1d..1e339cbbca9c2763a9abbc42e03cbddd1e89ed75 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -49,7 +49,6 @@ const Bold = core.Mark.create({
     addKeyboardShortcuts() {
         return {
             'Mod-b': () => this.editor.commands.toggleBold(),
-            'Mod-B': () => this.editor.commands.toggleBold(),
         };
     },
     addInputRules() {
diff --git a/dist/index.js b/dist/index.js
index a816a04e1b3cb41d9f0fb861ff448bd9fcfd5885..b5a8c882a586f0b923b60672084c50fbf934ea7f 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -45,7 +45,6 @@ const Bold = Mark.create({
     addKeyboardShortcuts() {
         return {
             'Mod-b': () => this.editor.commands.toggleBold(),
-            'Mod-B': () => this.editor.commands.toggleBold(),
         };
     },
     addInputRules() {
diff --git a/dist/index.umd.js b/dist/index.umd.js
index 7933c9a6e78536a7e7bc595ece5781e939767771..b735df6cdc6179556fa17345ea1e41c5d55ffc21 100644
--- a/dist/index.umd.js
+++ b/dist/index.umd.js
@@ -49,7 +49,6 @@
       addKeyboardShortcuts() {
           return {
               'Mod-b': () => this.editor.commands.toggleBold(),
-              'Mod-B': () => this.editor.commands.toggleBold(),
           };
       },
       addInputRules() {
