diff --git a/dist/index.mjs b/dist/index.mjs
index 00aaa666db615de4ab45d1d7955bd695caf7e284..1f888919908c770c3e2f219199f32904db7c3264 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -2440,19 +2440,6 @@ function addDeclarationTemplates(ctx, options) {
         return;
       }
       let path = resolveAlias(from);
-      if (!isAbsolute(path)) {
-        path = await tryResolveModule(from, nuxt.options.modulesDir).then(async (r2) => {
-          if (!r2) {
-            return r2;
-          }
-          const { dir, name } = parseNodeModulePath(r2);
-          if (!dir || !name) {
-            return r2;
-          }
-          const subpath = await lookupNodeModuleSubpath(r2);
-          return join(dir, name, subpath || "");
-        }) ?? path;
-      }
       if (isAbsolute(path)) {
         path = relative(join(nuxt.options.buildDir, "types"), path);
       }
