diff --git a/dist/app/components/nuxt-island.js b/dist/app/components/nuxt-island.js
index 4fe96ebfcbab2abbe1bccbd199e6dc8a2416ac64..668c1b42a4f1944cef1d533ff5f79c2822aadf55 100644
--- a/dist/app/components/nuxt-island.js
+++ b/dist/app/components/nuxt-island.js
@@ -1,6 +1,5 @@
 import { Fragment, Teleport, computed, createStaticVNode, createVNode, defineComponent, getCurrentInstance, h, nextTick, onBeforeUnmount, onMounted, ref, toRaw, watch, withMemo } from "vue";
 import { debounce } from "perfect-debounce";
-import { hash } from "ohash";
 import { appendResponseHeader } from "h3";
 import { randomUUID } from "uncrypto";
 import { joinURL, withQuery } from "ufo";
@@ -9,6 +8,68 @@ import { prerenderRoutes, useRequestEvent } from "../composables/ssr.js";
 import { injectHead } from "../composables/head.js";
 import { getFragmentHTML } from "./utils.js";
 import { appBaseURL, remoteComponentIslands, selectiveClient } from "#build/nuxt.config.mjs";
+
+// https://gist.github.com/hyamamoto/fd435505d29ebfa3d9716fd2be8d42f0?permalink_comment_id=3586164#gistcomment-3586164
+function hashString(s) {
+  let h = 0;
+  let i = s.length;
+
+  while (i > 0) {
+    h = (h << 5) - h + s.charCodeAt(--i) | 0;
+  }
+
+  return h.toString();
+}
+
+// https://github.com/shuding/stable-hash
+const table = new WeakMap();
+let counter = 0;
+function stableHash(arg) {
+  const type = typeof arg;
+  const constructor = arg && arg.constructor;
+  const isDate = constructor === Date;
+
+  if (Object(arg) === arg && !isDate && constructor !== RegExp) {
+    let result = table.get(arg);
+    if (result)
+      return result;
+    result = `${++counter}~`;
+    table.set(arg, result);
+    let index;
+
+    if (constructor === Array) {
+      // Array.
+      result = '@';
+      for (index = 0; index < arg.length; index++)
+        result += `${stableHash(arg[index])},`;
+
+      table.set(arg, result);
+    }
+    else if (constructor === Object) {
+      // Object, sort keys.
+      result = '#';
+      const keys = Object.keys(arg).sort();
+      // eslint-disable-next-line no-cond-assign
+      while ((index = keys.pop()) !== undefined) {
+        if (arg[index] !== undefined)
+          result += `${index}:${stableHash(arg[index])},`;
+      }
+      table.set(arg, result);
+    }
+    return result;
+  }
+  if (isDate)
+    return arg.toJSON();
+  if (type === 'symbol')
+    return arg.toString();
+  return type === 'string' ? JSON.stringify(arg) : `${arg}`;
+}
+
+function hash(arg) {
+  return hashString(stableHash(arg));
+}
+
+
 const pKey = "_islandPromises";
 const SSR_UID_RE = /data-island-uid="([^"]*)"/;
 const DATA_ISLAND_UID_RE = /data-island-uid(="")?(?!="[^"])/g;
