diff --git a/dist/app/components/nuxt-island.js b/dist/app/components/nuxt-island.js
index 5449cc44dca6006bef7805cac2600e295fec0731..c1095489368d7f06f3e12dacd2e590108d42351f 100644
--- a/dist/app/components/nuxt-island.js
+++ b/dist/app/components/nuxt-island.js
@@ -1,6 +1,5 @@
 import { Fragment, Teleport, computed, createStaticVNode, createVNode, defineComponent, getCurrentInstance, h, nextTick, onMounted, ref, toRaw, watch, withMemo } from "vue";
 import { debounce } from "perfect-debounce";
-import { hash } from "ohash";
 import { appendResponseHeader } from "h3";
 import { injectHead } from "@unhead/vue";
 import { randomUUID } from "uncrypto";
@@ -34,6 +33,67 @@ async function loadComponents(source = appBaseURL, paths) {
   }
   await Promise.all(promises);
 }
+
+// https://gist.github.com/hyamamoto/fd435505d29ebfa3d9716fd2be8d42f0?permalink_comment_id=3586164#gistcomment-3586164
+function hashString(s) {
+  let h = 0;
+  let i = s.length;
+
+  while (i > 0) {
+    h = (h << 5) - h + s.charCodeAt(--i) | 0;
+  }
+
+  return h.toString();
+}
+
+// https://github.com/shuding/stable-hash
+const table = new WeakMap();
+let counter = 0;
+function stableHash(arg) {
+  const type = typeof arg;
+  const constructor = arg && arg.constructor;
+  const isDate = constructor === Date;
+
+  if (Object(arg) === arg && !isDate && constructor !== RegExp) {
+    let result = table.get(arg);
+    if (result)
+      return result;
+    result = `${++counter}~`;
+    table.set(arg, result);
+    let index;
+
+    if (constructor === Array) {
+      // Array.
+      result = '@';
+      for (index = 0; index < arg.length; index++)
+        result += `${stableHash(arg[index])},`;
+
+      table.set(arg, result);
+    }
+    else if (constructor === Object) {
+      // Object, sort keys.
+      result = '#';
+      const keys = Object.keys(arg).sort();
+      // eslint-disable-next-line no-cond-assign
+      while ((index = keys.pop()) !== undefined) {
+        if (arg[index] !== undefined)
+          result += `${index}:${stableHash(arg[index])},`;
+      }
+      table.set(arg, result);
+    }
+    return result;
+  }
+  if (isDate)
+    return arg.toJSON();
+  if (type === 'symbol')
+    return arg.toString();
+  return type === 'string' ? JSON.stringify(arg) : `${arg}`;
+}
+
+function hash(arg) {
+  return hashString(stableHash(arg));
+}
+
 export default defineComponent({
   name: "NuxtIsland",
   props: {
diff --git a/dist/index.mjs b/dist/index.mjs
index 1275a0e7e6ae17c3f88cf6a20003ba3c19e58ef0..81e68f62b4ea7a21f1e402e7add945393d473dcd 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -2919,19 +2919,6 @@ function addDeclarationTemplates(ctx, options) {
         return;
       }
       let path = resolveAlias(from);
-      if (!isAbsolute(path)) {
-        path = await tryResolveModule(from, nuxt.options.modulesDir).then(async (r2) => {
-          if (!r2) {
-            return r2;
-          }
-          const { dir, name } = parseNodeModulePath(r2);
-          if (!dir || !name) {
-            return r2;
-          }
-          const subpath = await lookupNodeModuleSubpath(r2);
-          return join(dir, name, subpath || "");
-        }) ?? path;
-      }
       if (existsSync(path) && !await isDirectory$1(path)) {
         path = path.replace(SUPPORTED_EXTENSION_RE, "");
       }
