diff --git a/dist/index.mjs b/dist/index.mjs
index 240c7be41391c486bffd5c32132540ce87d4ce03..83fe1d30b88d15ed47349722d37cd0bf79a9d9c1 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -2436,19 +2436,6 @@ function addDeclarationTemplates(ctx, options) {
         return;
       }
       let path = resolveAlias(from);
-      if (!isAbsolute(path)) {
-        path = await tryResolveModule(from, nuxt.options.modulesDir).then(async (r2) => {
-          if (!r2) {
-            return r2;
-          }
-          const { dir, name } = parseNodeModulePath(r2);
-          if (!dir || !name) {
-            return r2;
-          }
-          const subpath = await lookupNodeModuleSubpath(r2);
-          return join(dir, name, subpath || "");
-        }) ?? path;
-      }
       if (isAbsolute(path)) {
         path = relative(join(nuxt.options.buildDir, "types"), path);
       }
