diff --git a/dist/app/composables/asyncData.js b/dist/app/composables/asyncData.js
index d7ecff5ec0c17891f9eb6da15c68d59fef57d951..0851dfc675f6c2df32367cc48854c9e4104cc8b8 100644
--- a/dist/app/composables/asyncData.js
+++ b/dist/app/composables/asyncData.js
@@ -20,8 +20,8 @@ export function useAsyncData(...args) {
   options.lazy = options.lazy ?? false;
   options.immediate = options.immediate ?? true;
   const nuxt = useNuxtApp();
-  const getCachedData = () => nuxt.isHydrating ? nuxt.payload.data[key] : nuxt.static.data[key];
-  const hasCachedData = () => getCachedData() !== void 0;
+  const getCachedData = () => nuxt.isHydrating ? toRef(nuxt.payload.data, key) : nuxt.static.data[key];
+  const hasCachedData = () => unref(getCachedData()) !== void 0;
   if (!nuxt._asyncData[key] || !options.immediate) {
     nuxt._asyncData[key] = {
       data: ref(getCachedData() ?? options.default()),
