diff --git a/dist/index.mjs b/dist/index.mjs
index bb6a08c3f04ddd0faf45f386c658a210290fe3e6..1565fda2c50d4aba787e29ce897eb6300cb2f46e 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -2455,7 +2455,7 @@ const TreeShakeComposablesPlugin = createUnplugin((options) => {
       const s = new MagicString(code);
       const strippedCode = stripLiteral(code);
       for (const match of strippedCode.matchAll(COMPOSABLE_RE_GLOBAL) || []) {
-        s.overwrite(match.index, match.index + match[0].length, `${match[1]} /*#__PURE__*/ false && ${match[2]}`);
+        s.overwrite(match.index, match.index + match[0].length, `${match[1]} false && /* @__PURE__ */ ${match[2]}`);
       }
       if (s.hasChanged()) {
         return {
