diff --git a/index.js b/index.js
index 36eb1f219a0844b48c6dbf57f5e910bd1a208ee1..a8fd0229ae665299ff08475aba8d7c4626ecbac8 100644
--- a/index.js
+++ b/index.js
@@ -1,19 +1,21 @@
-window.requestIdleCallback =
-    window.requestIdleCallback ||
-    function(cb) {
-        var start = Date.now();
-        return setTimeout(function() {
-            cb({
-                didTimeout: false,
-                timeRemaining: function() {
-                    return Math.max(0, 50 - (Date.now() - start));
-                },
-            });
-        }, 1);
-    };
+if (typeof window !== 'undefined') {
+    window.requestIdleCallback =
+        window.requestIdleCallback ||
+        function(cb) {
+            var start = Date.now();
+            return setTimeout(function() {
+                cb({
+                    didTimeout: false,
+                    timeRemaining: function() {
+                        return Math.max(0, 50 - (Date.now() - start));
+                    },
+                });
+            }, 1);
+        };
 
-window.cancelIdleCallback =
-    window.cancelIdleCallback ||
-    function(id) {
-        clearTimeout(id);
-    };
+    window.cancelIdleCallback =
+        window.cancelIdleCallback ||
+        function(id) {
+            clearTimeout(id);
+        };
+}
